### PR TITLE
Feature/histograms

### PR DESCRIPTION
**Pull request by AnmolDhaliwal00**
Histograms for proportions of answers have been implemented, as well as histograms which showcase various stats over time in relation to trial data.

### CODE REVIEW

File: app/src/main/java/xyz/kotlout/kotlout/view/fragment/ExperimentInfoFragment.java [hunk 1]
```diff
@@ -72,42 +93,19 @@ public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container,
...
-
-  }
-
-  private void FormatBinomial (BarChart trialInfo){
-
-  }
-
-  private void FormatNonNegMeasurement (BarChart histogram){
-    BarDataSet barDataSet = new BarDataSet(trialEntries, "Means");
+  private void FormatPlot (BarChart histogram, String dataType){
```
Comment (n30phyte) [inline app/src/main/java/xyz/kotlout/kotlout/view/fragment/ExperimentInfoFragment.java]: Fix naming, should be formatPlot

File: app/src/main/java/xyz/kotlout/kotlout/view/fragment/ExperimentInfoFragment.java [hunk 2]
```diff
@@ -159,6 +157,167 @@ private void FormatHistogram(BarChart histogram) {
     histogram.setScaleY(1);
   }
 
+  @Override
+  public void onExperimentLoaded() {
+    experiment = controller.getExperimentContext();
+    trialList = controller.getListTrials();
+    if (trialList.size() > 0){
+      switch (trialList.get(0).getClass().getName()){
```
Comment (n30phyte) [inline app/src/main/java/xyz/kotlout/kotlout/view/fragment/ExperimentInfoFragment.java]: Don't use reflection for this

File: app/src/main/java/xyz/kotlout/kotlout/model/experiment/trial/CountTrial.java [hunk 3]
```diff
@@ -23,4 +25,9 @@ public long getResult() {
   public void setResult(long result) {
     this.result = result;
   }
+
+  @Override
+  public int compareTo(CountTrial o) {
```
Comment (n30phyte) [inline app/src/main/java/xyz/kotlout/kotlout/model/experiment/trial/CountTrial.java]: Pls remove

File: app/src/main/java/xyz/kotlout/kotlout/model/experiment/trial/BinomialInfo.java [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+  public void incrementSuccess() {
+    this.success++;
+  }
+
+  public void incrementFailure() {
+    this.failure++;
+  }
+
+  public float successProportion(){
+    return (float) (this.success/(this.success + this.failure))*100;
```
Comment (n30phyte) [inline app/src/main/java/xyz/kotlout/kotlout/model/experiment/trial/BinomialInfo.java]: Run auto formatter

**Review by n30phyte — APPROVED** [review]
Aside from comments, lgtm

**Review by n30phyte — APPROVED** [review]

**Review by Mia2b — APPROVED** [review]
Seems to work fine.

Ship it!

**Pull request merged**